### DISCUSSION TITLE
How to use the "java_home" as runtime env

### DISCUSSION THREAD

**Discussion by Nonoas** [fvarrui/JavaPackager#402] [Q&A]
I want firstly use "java_home" to run the .exe, so that the user could not install jre for every javafx app.how to do it?

**Comment by fvarrui** [comment]
Hi @Nonoas!
You just have to not bundle a JRE with your app setting `bundleJre=false`. I guess that this should do the trick

**Reply by Nonoas** [reply]
It seems to no use, and I got a error like this 
![image](https://github.com/fvarrui/JavaPackager/assets/54799528/6ea3832e-afc9-41b2-a545-dc88a71a04ef)

**Reply by fvarrui** [reply]
Which JRE have you installed? Is JAVA_HOME env variable well defined?

**Reply by fvarrui** [reply]
Could you post here your JP config?

**Reply by Nonoas** [reply]
@fvarrui 
java version
```
java version "17" 2021-09-14 LTS
Java(TM) SE Runtime Environment (build 17+35-LTS-2724)
Java HotSpot(TM) 64-Bit Server VM (build 17+35-LTS-2724, mixed mode, sharing)
```
JP config
```groovy
tasks.register('packageMyApp', PackageTask) {
    dependsOn clean, jar
    // mandatory
    mainClass = mainClassName
    // optional
//    modules = ["java.base", "java.management", "java.net.http", "java.scripting", "java.sql",
//               "jdk.jsobject", "jdk.unsupported", "jdk.unsupported.desktop", "jdk.xml.dom"]
    bundleJre = false
    generateInstaller = false
    administratorRequired = false
    winConfig { createZipball = true }

}
```

**Reply by fvarrui** [reply]
Is JAVA_HOME env variable well defined?

**Reply by Nonoas** [reply]
> @fvarrui java version
> 
> ```
> java version "17" 2021-09-14 LTS
> Java(TM) SE Runtime Environment (build 17+35-LTS-2724)
> Java HotSpot(TM) 64-Bit Server VM (build 17+35-LTS-2724, mixed mode, sharing)
> ```
> 
> JP config
> 
> ```groovy
> tasks.register('packageMyApp', PackageTask) {
>     dependsOn clean, jar
>     // mandatory
>     mainClass = mainClassName
>     // optional
> //    modules = ["java.base", "java.management", "java.net.http", "java.scripting", "java.sql",
> //               "jdk.jsobject", "jdk.unsupported", "jdk.unsupported.desktop", "jdk.xml.dom"]
>     bundleJre = false
>     generateInstaller = false
>     administratorRequired = false
>     winConfig { createZipball = true }
> 
> }
> ```

this is output of my `java -version` , so JAVA_HOME must no problem

**Comment by fvarrui** [comment]
Sorry, but running `java -version` only proves that Java is defined in your PATH env variable. Please, run `echo %JAVA_HOME%` just to be sure.

Anyway, I'll try to reproduce your issue and tell you something asap
